proto/process/agent.proto

```diff
@@ -938,85 +947,112 @@ enum ResourceRequirementsType {
 message CronJobStatus {
-    repeated ObjectReference active = 1;
-    int64 lastScheduleTime = 2;
+	repeated ObjectReference active = 1;
+	int64 lastScheduleTime = 2;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/batch/v1beta1/types.go#L58
 message CronJob {
-    Metadata metadata = 1;
-    CronJobSpec spec = 2;
-    CronJobStatus status = 3;
-    bytes yaml = 4;
-    repeated string tags = 5;
+	Metadata metadata = 1;
+	CronJobSpec spec = 2;
+	CronJobStatus status = 3;
+	bytes yaml = 4;
+	repeated string tags = 5;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L119
 message DaemonSetSpec {
-    repeated LabelSelectorRequirement selectors = 1;
-    string deploymentStrategy = 2;
-    string maxUnavailable = 3; // int or str
-    int32 minReadySeconds = 4;
-    int32 revisionHistoryLimit = 5;
+	repeated LabelSelectorRequirement selectors = 1;
+	string deploymentStrategy = 2;
+	string maxUnavailable = 3; // int or str
+	int32 minReadySeconds = 4;
+	int32 revisionHistoryLimit = 5;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L152
 message DaemonSetStatus {
-    int32 currentNumberScheduled = 1;
-    int32 numberMisscheduled = 2;
-    int32 desiredNumberScheduled = 3;
-    int32 numberReady = 4;
-    int32 updatedNumberScheduled = 6;
-    int32 numberAvailable = 7;
-    int32 numberUnavailable = 8;
+	int32 currentNumberScheduled = 1;
+	int32 numberMisscheduled = 2;
+	int32 desiredNumberScheduled = 3;
+	int32 numberReady = 4;
+	int32 updatedNumberScheduled = 6;
+	int32 numberAvailable = 7;
+	int32 numberUnavailable = 8;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L66
 message DaemonSet {
-    Metadata metadata = 1;
-    DaemonSetSpec spec = 2;
-    DaemonSetStatus status = 3;
-    bytes yaml = 4;
-    repeated string tags = 5;
+	Metadata metadata = 1;
+	DaemonSetSpec spec = 2;
+	DaemonSetStatus status = 3;
+	bytes yaml = 4;
+	repeated string tags = 5;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L586
+message StatefulSetSpec {
+	int32 replicas = 1;
+	repeated LabelSelectorRequirement selectors = 2;
+	string serviceName = 3;
+	string podManagementPolicy = 4;
+	string updateStrategy = 5;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L647
+message StatefulSetStatus {
+	int32 replicas = 1;
+	int32 readyReplicas = 2;
+	int32 currentReplicas = 3;
+	int32 updatedReplicas = 4;
+	int32 collisionCount = 5;
```

Comment:
AFAICT we never collected the `collisionCount` (Deployment and Daemonsets). Do we need that field? Seems to be something more internal

Reply:
+ do we want to add a comment in the PR description, I know you already did in slack, why we don't collect specific fields? I think this makes remembering the reason later easier 👍